zio-http/src/test/scala/zhttp/service/ServerSpec.scala

```diff
@@ -170,6 +171,17 @@ object ServerSpec extends HttpRunnableSpec {
         val res  = Http.fromStream(ZStream.fromFile(Paths.get(path))).requestBodyAsString()
         assertM(res)(equalTo("abc\nfoo"))
       } +
+      testM("response streaming memory leak") {
+        for {
+          _     <- Http
+            .fromData(HttpData.fromStream(ZStream.fail(new Error)))
+            .requestBodyAsString(content = "test")
+          alloc <- getStart.map(_.allocator.get)
+          ah    <- getActiveHeapBuffers(alloc)
+          ad    <- getActiveDirectBuffers(alloc)
+
+        } yield assertTrue(ah + ad == 0L)
```

Comment:
This check should either happen either in the end or just before the server stops.